demo/sizzle.sysl

```diff
@@ -0,0 +1,318 @@
+        @description ="this is a description of Authenticate"
+        # TODO: how to present database interaction here?
+        if authenticated:
+          return 200 <: MegaDatabase.Empty
+        else:    
+          return 401 <: UnauthorizedError
+
+    CustomerProfile(customer_id <: int):
+        # TODO: database interaction
+        return ok <: Customer
+
+    UpdatePassword(customer_id <: int, old <: string, new <: string):
+        # DB update
+        return ok
+
+    !type Customer:
+        @description="Empty Type"
+        first_name <: string
+        middle_name <: string
+        last_name <: string
+        phone <: string
+        email <: string
+
+    !type NewCustomerRequest:
+        first_name <: string
+        middle_name <: string
+        last_name <: string
+        phone <: string
+        email <: string
+        password <: string
+
+    !type UnauthorizedError:
+        error_msg <: string
+
+ProductServer:
+    @package="BFF"
+    @description =: 
+        | This server handles all the product
+        | related endpoints.
+
+    Menu:
+        # TODO: DB/cache op
+        return ok <: Products
+
+    !alias Products:
+        sequence of Product
+
+    !type Product:
+        @description="Product information"
+        id <: int
+        name <: string
+        image <: string
+        details <: string
+        price <: int
+
+
+OrderServer:
+    @package="BFF"
+    @description =: 
+        | This server handles all the order
+        | related endpoints.
+
+    Order(req <: OrderRequest):
+        if order_id is nil:
+            # TODO: DB op create a new order
+            return ok <: Order
+        else:
+            # TODO: add/update OrderProduct for the 
+            # existing order
+            return ok <: Order
+
+    UpdateOrderStatus(order_id <: int, status <: int):
+        # TODO: DB op get order by order_id
+        # state machine
+        return ok <: Order
+
+    Review(score <: int, comment <: string):
+        # TODO: DB op
+        return ok <: Order
+
+    !type OrderRequest:
+        order_id <: int
+        product_id <: int
+        quantity <: int
+
+    !type Order:
+        @description="Customer order information"
+        id <: int
+        status <: OrderStatus
+        total_price <: int
+        items <: sequence of OrderProduct
+        paid <: bool
+        review_score <: int
+        review_comment <: string
+
+    !type OrderProduct:
+        @description="Order items"
+        product_id <: int
+        quantity <: int
+        comments <: string
+
+    !enum OrderStatus:
+        created: 1
+        placed: 2
+        shipped: 3
+        delivered: 4
+    
+
+PaymentServer:
+    @package="BFF"
+    @description =: 
+        | This server handles all the payment related endpoints.
+
+    Pay:
+        if processor_type == "visa":
+            Visa <- Pay
+        else if processor_type == "mastercard":
+            MasterCard <- POST /pay
```

Comment:
Seems like the diagram isn't showing the call to MasterCard, which is interesting
![image](https://user-images.githubusercontent.com/25007585/88146156-d2821300-cc3e-11ea-9cfc-65a1a62db04f.png)

Reply:
Yes that is a new issue. Also this integration diagram should show "Dine-in Customer -> Dashboard -> PaymentServer" but it doesn't.
![image](https://user-images.githubusercontent.com/13619421/88147478-9b146600-cc40-11ea-8ac9-c2721328467f.png)